### ISSUE TITLE
Amplify doesn't count custom events if you're notifying the lib using new instances

### ISSUE THREAD

**Issue by iAlex97** [stkent/amplify#181]
First of all, I'd like to thank you about this cool library. I've stumbled upon it just today and the customization level just baffles me.

I did, however run into a small issue when trying to implement a custom event that I would have liked to happen a minimum number of times before showing the prompt.

According to the docs, I created a new class which implements `IEvent`:

```
public class ButtonClickedEvent implements IEvent {
    @NonNull
    @Override
    public String getTrackingKey() {
        return "KEY_BUTTON_CLICKED";
    }
}
```
and registered it along the other "relevant" events:
```
Amplify.initSharedInstance(this)
                .setPositiveFeedbackCollectors(new GooglePlayStoreFeedbackCollector())
                .setCriticalFeedbackCollectors(new DefaultEmailFeedbackCollector("someone@example.com"))
                .addTotalEventCountRule(PromptInteractionEvent.USER_GAVE_POSITIVE_FEEDBACK,
                        new MaximumCountRule(1))
                .addTotalEventCountRule(new ButtonClickedEvent(), new MinimumCountRule(3)); // my custom event
```

I wired up that event to a button click event and logged the result of the `shouldPrompt` method:
```
Amplify.getSharedInstance().notifyEventTriggered(new ButtonClickedEvent());

boolean condition = Amplify.getSharedInstance().shouldPrompt();

Timber.d("Should show rate notification => %b", condition);
```

which resulted in a quite strange logging of event (my custom event was dispatched but never incremented):
```
D/Amplify: KEY_BUTTON_CLICKED event triggered
D/Amplify: No tracked value for total count of USER_GAVE_POSITIVE_FEEDBACK event
D/Amplify: No tracked value for total count of KEY_BUTTON_CLICKED event
D/Amplify: Blocking feedback because of MinimumCountRule with minimum required count of 3 associated with KEY_BUTTON_CLICKED event
D/MainActivity: Should show rate notification => false
D/Amplify: KEY_BUTTON_CLICKED event triggered
D/Amplify: No tracked value for total count of USER_GAVE_POSITIVE_FEEDBACK event
D/Amplify: No tracked value for total count of KEY_BUTTON_CLICKED event
D/Amplify: Blocking feedback because of MinimumCountRule with minimum required count of 3 associated with KEY_BUTTON_CLICKED event
D/MainActivity: Should show rate notification => false
```

investigating the propagation of that event, I found out that it's getting rejected at the following line: https://github.com/stkent/amplify/blob/a32a62e3deb1f5cac7c2c4c288bd3ae98dd67b8a/amplify/src/main/java/com/github/stkent/amplify/tracking/managers/BaseEventsManager.java#L86

which makes sense because the HashMap doesn't have the newly reference included.

TL;DR:
I solved this issue by using a static instance when registering and notifying about "custom" events.

I think this should be mentioned in the documentation to avoid further confusion.

**Comment by stkent** [comment]
Hi @iAlex97!

Glad you are finding the library useful :) Thanks for the report - this definitely looks like a bug and I have an idea as to how to fix it so you are not required to use a static instance. I'll make that fix today and let you know when a new version is available!

**Comment by iAlex97** [comment]
No worries @stkent :)
Personally, I can use it as it is without too much hassle, just thought this might help others too.

**Comment by stkent** [comment]
@iAlex97 yeah, this was really likely to trip up others given the combination of undocumented requirement to use the same event instance and a README example that creates a new event instance on the fly ;)

This should now be fixed in release 2.2.1, where creating new event instances on the fly when notifying should work just fine!
